drivers/flash/spi_nor.c

```diff
@@ -576,55 +610,64 @@ static int mxicy_rdcr(const struct device *dev)
  *
  * @return 0 on success or a negative error code.
  */
-static int mxicy_wrcr(const struct device *dev,
-			uint16_t cr)
+static int mxicy_wrcr(const struct device *dev, uint16_t cr)
 {
+	const struct spi_nor_config *cfg = dev->config;
+	int ret = -ENOSYS;
 	/* The configuration register bytes on the Macronix MX25R devices are
 	 * written using the Write Status Register command where the configuration
 	 * register bytes are written as two extra bytes after the status register.
 	 * First read out the current status register to preserve the value.
 	 */
-	int sr = spi_nor_rdsr(dev);
 
-	if (sr < 0) {
-		LOG_ERR("Read status register failed: %d", sr);
-		return sr;
-	}
+	if (cfg->mxicy_mx25r_power_mode_exist) {
+		int sr = spi_nor_rdsr(dev);
 
-	int ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_WREN);
+		if (sr < 0) {
+			LOG_ERR("Read status register failed: %d", sr);
+			return sr;
+		}
 
-	if (ret == 0) {
-		uint8_t data[] = {
-			sr,
-			cr & 0xFF,	/* Configuration register 1 */
-			cr >> 8		/* Configuration register 2 */
-		};
+		ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_WREN);
+
+		if (ret == 0) {
+			uint8_t data[] = {
+				sr, cr & 0xFF, /* Configuration register 1 */
+				cr >> 8		/* Configuration register 2 */
```

Comment:
This looks more confusing than the previous one because you're combining sr with cr, writing to sr and cr is writing to 2 registers not one